tokio-macros/src/select.rs

```diff
@@ -1,11 +1,16 @@
-use proc_macro::TokenStream;
+use proc_macro::{TokenStream, TokenTree};
 use proc_macro2::Span;
 use quote::quote;
 use syn::Ident;
 
 pub(crate) fn declare_output_enum(input: TokenStream) -> TokenStream {
-    // passed in is: `!!!!` with one `!` per branch
-    let branches = input.into_iter().count();
+    // passed in is: `(_ _ _)` with one `_` per branch
+    let branches = match input.into_iter().next() {
+        Some(TokenTree::Group(group)) => {
+            group.stream().into_iter().count()
+        }
+        _ => panic!("unexpected macro input")
```

Comment:
is this an internal bug or does it mean that the input to the select macro was bad? if the latter, should this generate a nicer diagnostic for the user?

Reply:
it would be an internal bug.

Reply:
I changed the input to the internal macro to be more future proof.